lib/Core/Server/Transmitter/MessageFormatter.php

```diff
@@ -29,4 +29,24 @@ public function write(Message $message): string
             $body
         ]);
     }
+
+    /**
+     * Normalize a message before being serialized by recursively applying array_filter.
+     *
+     * @param mixed $message
+     *
+     * @return mixed
+     */
+    private function normalize($message)
```

Comment:
return type `array` ?

Reply:
Can't because we will also accept scalar (recursion) and return them.